game/aamp_v2.ksy

```diff
@@ -0,0 +1,203 @@
+    - bphysics
+    - bphyslayer
+    - bphysmaterial
+    - bphyssb
+    - bphyssubmat
+    - bptclconf
+    - brecipe
+    - brgbw
+    - brgcon
+    - brgconfig
+    - brgconfiglist
+    - bsfbt
+    - bsft
+    - bshop
+    - bumii
+    - bvege
+    - bactcapt
+  xref:
+    zeldamods: AAMP
+  endian: le
+seq:
+  - id: header
+    type: header
+    doc-ref: https://zeldamods.org/wiki/AAMP#Header
+  - id: parameter_lists
+    type: parameter_list
+    repeat: expr
+    repeat-expr: header.num_lists
+    doc-ref: https://zeldamods.org/wiki/AAMP#Parameter_list
+  - id: parameter_objects
+    type: parameter_object
+    repeat: expr
+    repeat-expr: header.num_objects
+    doc-ref: https://zeldamods.org/wiki/AAMP#Parameter_object
+  - id: parameters
+    type: parameter
+    repeat: expr
+    repeat-expr: header.num_parameters
+    doc-ref: https://zeldamods.org/wiki/AAMP#Parameter
+  - id: data_section
+    size: header.data_section_size
+    doc-ref: https://zeldamods.org/wiki/AAMP#Section_order
+  - id: string_section
+    size: header.string_section_size
+    doc-ref: https://zeldamods.org/wiki/AAMP#Section_order
+  - id: unknown_uint32_section
+    type: u1
+    repeat: eos
+    doc: May be unused.
+    doc-ref: https://zeldamods.org/wiki/AAMP#Section_order
+types:
+  header:
+    seq:
+    - id: magic
+      contents: AAMP
+    - id: version
+      type: u1
+      doc: Should be "2"
+    - id: flags
+      type: u1
+      repeat: expr
+      repeat-expr: 7
+      doc: TODO: Flags (LittleEndian: 1 << 0, UTF8: 1 << 1)
+    - id: file_size
+      type: u4
+    - id: pio_verision
+      type: u4
+    - id: pio_offset
+      type: u4
+    - id: num_lists
+      type: u4
+    - id: num_objects
+      type: u4
+    - id: num_parameters
+      type: u4
+    - id: data_section_size
+      type: u4
+    - id: string_section_size
+      type: u4
+    - id: unknown
+      size: 4
+    - id: pio_type
+      type: str
+      size: 4
+      encoding: utf-8
+      doc: typically "xml"
+  parameter_list:
+    seq:
+      - id: name_crc32
+        type: u4
+      - id: child_lists
+        type: u4
+        doc: Number of child lists
+      - id: child_objects
+        type: u4
+    instances:
+      child_lists_offset:
+        value: child_lists
+      num_child_lists:
+        value: child_lists >> 16
+  parameter_object:
+    seq:
+      - id: name_crc32
+        type: u4
+      - id: data
+        type: u4
+    instances:
+      child_offset:
+        value: data
+      num_child_params:
+        value: data >> 16
+  parameter:
+    seq:
+      - id: name_crc32
+        type: u4
+      - id: data
+        type: u4
+    instances:
+      data_offset:
+        value: data
+        doc:  Offset to data, divided by 4 and relative to parameter start.
+      parameter_type:
+        value: data >> 24
+        enum: parameter_type
+        doc-ref: https://zeldamods.org/wiki/AAMP#ParameterType
+enums:
+  parameter_type:
+    0:  bool
+    1:  f32
```

Comment:
Just a note:
https://github.com/KOLANICH/kaitai_struct_formats/blob/ieee754_float/common/ieee754_float/f32be.ksy
https://github.com/KOLANICH/kaitai_struct_formats/blob/ieee754_float/common/ieee754_float/f32le.ksy

may be helpful somewhen in future

Reply:
I don't understand this. This is just a lookup table for the types that Nintendo uses.
It's laid out on this page:
https://zeldamods.org/wiki/AAMP#ParameterType

Reply:
> I don't understand this. This is just a lookup table for the types that Nintendo uses.
> It's laid out on this page:
> https://zeldamods.org/wiki/AAMP#ParameterType

@KOLANICH has just probably thought that `f32` means a 32-**byte** floating-point number, but it is a normal 32-_bit_ float known as `f4` in Kaitai Struct.

@halotroop2288 Perhaps you could rename it to `f4` then to avoid the confusion? If you think that the original identifier should be kept, you can add it to the `-orig-id` key:

```ksy
    0:  bool
    1:
      id: f4
      -orig-id: f32
```